sway/tree/container.c

```diff
@@ -1609,3 +1620,69 @@ bool container_is_scratchpad_hidden_or_child(struct sway_container *con) {
 	con = container_toplevel_ancestor(con);
 	return con->scratchpad && !con->workspace;
 }
+
+static bool is_parallel(enum sway_container_layout first,
+		enum sway_container_layout second) {
+	switch (first) {
+	case L_TABBED:
+	case L_HORIZ:
+		return second == L_TABBED || second == L_HORIZ;
+	case L_STACKED:
+	case L_VERT:
+		return second == L_STACKED || second == L_VERT;
+	default:
+		return false;
+	}
+}
+
+static bool container_is_squashable(struct sway_container *con,
+		struct sway_container *child) {
+	enum sway_container_layout gp_layout = con->parent ?
+		con->parent->layout : con->workspace->layout;
+	enum sway_container_layout con_layout = con->layout;
```

Comment:
Hmm, is there a reason why these are stored in variables here? `is_parallel` is pure, and the logic below mixes `con->layout` with `con_layout`, and `child->layout` with `child_layout`.